ksql-common/src/main/java/io/confluent/ksql/schema/ksql/FormatOptions.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.ksql;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+public final class FormatOptions {
+
+  private final Predicate<String> reservedWordPredicate;
+
+  public static FormatOptions none() {
+    return new FormatOptions(word -> true);
+  }
+
+  /**
+   * Construct instance.
+   *
+   <p>The {@code reservedWordPredicate} allows code that lives in the common module
+   * to be wired up to the set of reserved words defined in the parser module. Wire this up to
+   * {@code ParserUtil::isReservedWord}.
+   *
+   * <p>If using this type in a module that does <i>not</i> have access to the parser, then the
+   * <i>safest</i> option is to pass in predicate that always returns true, which will always
```

Comment:
is this really the safest option? As you mentioned somewhere, this will make the field name case sensitive - which may not be what we want either. I think it might be best to just omit this guideline and always recommend piping through a predicate that can determine whether or not a word is reserved

Reply:
If only that were always possible!  Our code already makes sure everything is upper case, so quoting isn't an issue.  

Once we have case-sensitivity, we'll actually need to quote if things are reserved or not upper case.

TBH, just quoting everything is the best approach / safetest. Deciding to not quote just makes things more pretty.